src/PolynomialFunction.java

```diff
@@ -3,15 +3,18 @@
 import java.util.*;
 
 public class PolynomialFunction implements ParametricFunction {
-
-    private double[] coefficients;
     public String functionString;
+    private double[] coefficients;
+    public List<Double> roots;
+    public List<Double> extremePoints;
     private static final Pattern TERM_PATTERN = Pattern.compile("([-+]?\\s*\\d*\\.?\\d*(?:/\\d+)*)?x(\\^(-?\\d+))?|([-+]?\\s*\\d+(/\\d+)?)");
 
-    public PolynomialFunction(String polynomialString) {
+    public PolynomialFunction(String polynomialString, double start, double end) {
```

Comment:
a PolynomialFunction should not have a 'start' and 'end'

Reply:
So is the naming the problem? would something like leftBorder and rigthBorder  or similar be better?

Reply:
the function should be oblivious to its 'start' and 'end'. This also belongs in the GraphPanel imo.

Reply:
Done